server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexAction.java

```diff
@@ -0,0 +1,653 @@
+            setName(in.readString());
+            this.indices = in.readStringArray();
+        }
+
+        ResolvedAlias(String name, String[] indices) {
+            super(name);
+            this.indices = indices;
+        }
+
+        public ResolvedAlias copy(String newName) {
+            return new ResolvedAlias(newName, indices);
+        }
+
+        public String[] getIndices() {
+            return indices;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(getName());
+            out.writeStringArray(indices);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(NAME_FIELD.getPreferredName(), getName());
+            if (indices.length > 0) {
+                builder.array(INDICES_FIELD.getPreferredName(), indices);
+            }
+            builder.endObject();
+            return builder;
+        }
+    }
+
+    public static class ResolvedDataStream extends ResolvedIndexAbstraction implements Writeable, ToXContentObject {
+
+        private static final ParseField BACKING_INDICES_FIELD = new ParseField("backing_indices");
+        private static final ParseField TIMESTAMP_FIELD = new ParseField("timestamp_field");
+
+        private final String[] backingIndices;
+        private final String timestampField;
+
+        ResolvedDataStream(StreamInput in) throws IOException {
+            setName(in.readString());
+            this.backingIndices = in.readStringArray();
+            this.timestampField = in.readString();
+        }
+
+        ResolvedDataStream(String name, String[] backingIndices, String timestampField) {
+            super(name);
+            this.backingIndices = backingIndices;
+            this.timestampField = timestampField;
+        }
+
+        public ResolvedDataStream copy(String newName) {
+            return new ResolvedDataStream(newName, backingIndices, timestampField);
+        }
+
+        public String[] getBackingIndices() {
+            return backingIndices;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(getName());
+            out.writeStringArray(backingIndices);
+            out.writeString(timestampField);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(NAME_FIELD.getPreferredName(), getName());
+            builder.array(BACKING_INDICES_FIELD.getPreferredName(), backingIndices);
+            builder.field(TIMESTAMP_FIELD.getPreferredName(), timestampField);
+            builder.endObject();
+            return builder;
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
```

Comment:
maybe add a parsing unit test for the response class? (extending `AbstractSerializingTestCase` and defining a parser in the test)